File: 1746.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        c = defaultdict(list)
        N = len(s)
        for i in range(N):
            c[s[i]].append(i)
            
        res = -1
        for i, v in c.items():
            for i in range(1,len(v)):
                res = max(res, v[i] - v[i-1] - 1)
        
        return res
                
        

File: 1746.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        unordered_map<char, int> c;
        int res = -1;
        
        for(int i = 0; i < s.size(); i++){
            if(c.find(s[i]) != c.end())
                res = max(res, i - c[s[i]]- 1);
            else
                c[s[i]] = i;
        }
        return res;
    }
};